internal/api/validator/validator.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright (c) 2024 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+// Options is a struct that contains options for the validator.
+type Options struct {
+	// JobNameSanitizer is a replacer that sanitizes job names before name
+	// validation.
+	JobNameSanitizer *strings.Replacer
+}
+
+// Validator validates API request payloads.
+type Validator struct {
+	jobNameSanitizer *strings.Replacer
+}
+
+func New(opts Options) *Validator {
+	jobNameSanitizer := opts.JobNameSanitizer
+	if jobNameSanitizer == nil {
+		jobNameSanitizer = strings.NewReplacer("_", "", ":", "", "-", "")
```

Comment:
```suggestion
		jobNameSanitizer = strings.NewReplacer("_", "", ":", "", "-", "", " ", "")
```
if we add this I think we fix a discord user issue where they saw an err due to the job name (reminder/actor I believe) having a space in it.

Reply:
Thanks, makes sense to me- instead of adding it here (to an already bloated PR), I've made [this issue](https://github.com/diagridio/go-etcd-cron/issues/46) so we can track it separately and be intentional about the test cases we are adding with a new PR. There is also the open question about our current key segment concatenation and breaking changes https://github.com/diagridio/go-etcd-cron/pull/45

Reply:
sure thing.